### DISCUSSION TITLE
How to create a Custom Formula that returns an Array to fill in multiple cells

### DISCUSSION THREAD

**Discussion by qcurtemanjc** [handsontable/hyperformula#1177] [API Questions]
Hello,

I am working on creating a new SPLIT formula but I am running in to an issue. I think I have followed all of the documentation correctly on how to create a custom formula that returns an Array, however no matter what I do, I always get a `#SPILL!` error when I call my function from the `arraySizeMethod` I have defined. I even return an `ArraySize` object where I have explicitly set the width/height to what it should be and it still gives me this error. Here is an example of what I am seeing: https://codepen.io/qcurtemanjc/pen/oNPedpB 

Let me know if you need anything else from me, thank you!

P.S.
Can we also make the `SimpleRangeValue` class a public class like you guys recently did with `ArraySize` class? That class is needed to return the values as an array from the custom formula but I haven't been able to find it (I could have missed it as well so let me know if I'm mistaken) thanks!

**Comment by wojciechczerniak** [comment]
You have values in the cells around the formula so it cannot "spill" to them. Change `''` to `null` should help with that

**Reply by qcurtemanjc** [reply]
@wojciechczerniak Thanks for the quick response! Changing the values from `''` to `null` did seem to fix the `spill` issue.

The next issue I'm having is with `SimpleRangeValue`. I copied over the `ArraySize` and `SimpleRangeValue` classes from Hyperformula into my Codepen example to get rid of any errors but I'm still not getting the result expected from the custom function https://codepen.io/qcurtemanjc/pen/oNPedpB

I would expect that cell `A2` would have the value `1` and cell `B2` would have the value `2` but it just blank.

Thank you!

**Reply by wojciechczerniak** [reply]
IMO copying code won't help. Internally HF checks "instanceof" and your 'SimpleRangeValue' is different than internal 'SimpleRangeValue'. Range will not be displayed although it's evaluated correctly.

**Reply by qcurtemanjc** [reply]
I see, is there a possible workaround for the current time until `SimpleRangeValue` has been added as a public class?

**Reply by wojciechczerniak** [reply]
Looks like it's exported: https://github.com/handsontable/hyperformula/blob/f829f4c57519fffab0ce3a2e3d27182b73cb9a56/src/index.ts#L175 but not exposed on Hyperformula object. Should work with ES6 modules or some bundler/build system.

**Reply by AMBudnik** [reply]
Hi guys! We've already marked the issue with `SimpleRangeValue` at https://github.com/handsontable/hyperformula/issues/1178 and will inform you as soon as we fix it.

**Reply by AMBudnik** [reply]
ps @qcurtemanjc I've just tested your code locally, and it works well. Here's the project zipped https://we.tl/t-Q4Z2GZES1G

**Reply by AMBudnik** [reply]
I just wanted to make a quick note that HyperFomula 2.4.0 has launched, and the issue https://github.com/handsontable/hyperformula/issues/1178 is fixed.
